client.go

```diff
@@ -1907,6 +2017,11 @@ func (r *Receiver) Receive(ctx context.Context) (*Message, error) {
 	// delivered regardless of whether the link has been closed.
 	select {
 	case msg := <-r.link.messages:
+		// we remove the message from pending as soon as it's popped off the channel
+		// This makes the pending count the same as messages buffer count
+		// and keeps the behavior the same as before the pending messages tracking was introduced
+		defer r.link.deletePending(&msg)
```

Comment:
defering the deletePending call effectively makes the Receive() func behave the same as before. it removes the message from the map as soon as it's popped from the channel.